skimage/exposure/_adapthist.py

```diff
@@ -76,19 +70,12 @@ def equalize_adapthist(image, ntiles_x=None, ntiles_y=None, clip_limit=0.01,
     image = img_as_uint(image)
     image = rescale_intensity(image, out_range=(0, NR_OF_GREY - 1))
 
-    if ntiles_x is not None or ntiles_y is not None:
-        warn('`ntiles_*` have been deprecated in favor of '
-             '`kernel_size`.  The `ntiles_*` keyword arguments '
-             'will be removed in v0.14', skimage_deprecation)
-
-    if kernel_size is None:
-        ntiles_x = ntiles_x or 8
-        ntiles_y = ntiles_y or 8
-        kernel_size = (np.round(image.shape[0] / ntiles_y),
-                       np.round(image.shape[1] / ntiles_x))
+    if kwargs:
+        if 'ntiles_x' in kwargs or 'ntiles_y' in kwargs:
+            raise ValueError('`ntiles_*` have been deprecated in favor of `kernel_size`')
 
     if isinstance(kernel_size, numbers.Number):
-        kernel_size = (kernel_size, kernel_size)
+        raise ValueError('`kernel_size` must be a 2-tuple instead of an integer')
```

Comment:
Instead of this, let's broadcast up to the image dimensionality, e.g., with no loss of n-dimensional generality:

``` python
kernel_size = (kernel_size,) * image.ndim
```

Also will need to change the docstring to reflect as much (integer or iterable), explaining that if an iterable is passed it should have the same length as the image dimensionality. Checking this is true would be a good check to add, with an error raised, as that does mean something is wrong.